metaflow/plugins/argo/argo_live_run.py

```diff
@@ -0,0 +1,248 @@
+                raise Exception("Error - Unable to trigger run")
+            time.sleep(1)
+        else:
+            # TODO: add specificity to exceptions (see AIP-6470).
+            # MetaflowException (or a child exception in file)
+            # or TimeoutError would be more specific.
+            raise Exception("Failed to begin running")
+
+        # optional wait for argo workflow to finish running
+        if self._wait:
+            print("Now we will wait for the flow to finish")
+
+            start_time = time.time()
+            loop_counter = 0
+            while self._wait_timeout * 60 > time.time() - start_time:
+                if not self.is_running:
+                    break
+                if loop_counter % 12 == 0:
+                    print(
+                        f"Time waited: {int((time.time() - start_time)/60)}"
+                        f" minutes out of a possible {self._wait_timeout}"
+                    )
+                loop_counter += 1
+                time.sleep(5)
+            else:
+                print("Run timed out.")
+                # TODO: add specificity to exceptions (see AIP-6470)
+                raise Exception("Wait Timeout")
+
+            success_statement = (
+                "successfully!!!" if self.successful else "unsuccessfully."
+            )
+            print(f"\nRun completed {success_statement}")
+
+        else:
+            print("\nNot waiting for run to finish")
+
+    @property
+    def _status(self) -> str:
+        if self._cached_status in ["Error", "Failed", "Succeeded"]:
+            return self._cached_status
+
+        workflow = self._argo_client.get_workflow(self._argo_run_id)
+        if workflow.get("status"):
+            self._cached_status = workflow["status"].get("phase")
+
+        return self._cached_status
+
+    @property
+    def has_triggered(self) -> bool:
+        if self._has_triggered:
+            return True
+
+        elif self._status and self._status != "Error":
+            self._has_triggered = True
+
+        return self._has_triggered
+
+    @property
+    def is_running(self) -> bool:
+        return self._status == "Running"
+
+    @property
+    def _error(self) -> bool:
+        return self._status == "Error"
+
+    @property
+    def successful(self) -> bool:
+        return self._status == "Succeeded"
+
+    @property
+    def failed_steps(self) -> list:
+        if self.is_running or self.successful:
+            return []
+
+        failed_steps = []
+
+        workflow = self._argo_client.get_workflow(self._argo_run_id)
+        nodes_info = workflow["status"]["nodes"]
+        for node in nodes_info:
+            if (
+                nodes_info[node]["phase"] == "Failed"
+                and nodes_info[node]["type"] == "Pod"
+            ):
+                failed_steps.append(nodes_info[node]["templateName"])
+
+        self._failed_steps = failed_steps
+
+        return self._failed_steps
```

Comment:
If failed_steps are regenerated every time user call `failed_step` there's no need storing it as an instance variable `self._failed_steps`.

Reply:
done